modules/SofaMiscFem/src/SofaMiscFem/TriangularFEMForceField.inl

```diff
@@ -196,7 +196,7 @@ void TriangularFEMForceField<DataTypes>::initSmall(int i, Index&a, Index&b, Inde
 template <class DataTypes>
 void TriangularFEMForceField<DataTypes>::initLarge(int i, Index&a, Index&b, Index&c)
 {
-    type::vector<TriangleInformation>& triangleInf = *(triangleInfo.beginEdit());
+    type::vector<TriangleInformation>& triangleInf = *(triangleInfo.beginWriteOnly());
```

Comment:
I suggested the following replacement to use RAII instead of begin/end pairs (everywhere)
```cpp
auto triangleInf = getWriteOnlyAccessor(triangleInfo)
```

Reply:
I tried but it gives me this error:
```
16>C:\projects\sofa-src\modules\SofaMiscFem\src\SofaMiscFem/TriangularFEMForceField.inl(206,140): error C2039: 'size': is not a member of 'sofa::helper::WriteOnlyAccessor<sofa::component::topology::TopologyData<sofa::topology::Triangle,sofa::type::vector<sofa::component::forcefield::TriangularFEMForceField<sofa::defaulttype::Vec3Types>::TriangleInformation,sofa::type::CPUMemoryManager<T>>>,void>'
16>        with
16>        [
16>            T=sofa::component::forcefield::TriangularFEMForceField<sofa::defaulttype::Vec3Types>::TriangleInformation
16>        ]
16>C:\projects\sofa-src\SofaKernel\modules\SofaHelper\src\sofa/helper/accessor.h(144): message : see declaration of 'sofa::helper::WriteOnlyAccessor<sofa::component::topology::TopologyData<sofa::topology::Triangle,sofa::type::vector<sofa::component::forcefield::TriangularFEMForceField<sofa::defaulttype::Vec3Types>::TriangleInformation,sofa::type::CPUMemoryManager<T>>>,void>'
16>        with
16>        [
16>            T=sofa::component::forcefield::TriangularFEMForceField<sofa::defaulttype::Vec3Types>::TriangleInformation
16>        ]
16>C:\projects\sofa-src\modules\SofaMiscFem\src\SofaMiscFem/TriangularFEMForceField.inl(211,48): error C2676: binary '[': 'sofa::helper::WriteOnlyAccessor<sofa::component::topology::TopologyData<sofa::topology::Triangle,sofa::type::vector<sofa::component::forcefield::TriangularFEMForceField<sofa::defaulttype::Vec3Types>::TriangleInformation,sofa::type::CPUMemoryManager<T>>>,void>' does not define this operator or a conversion to a type acceptable to the predefined operator
16>        with
16>        [
16>            T=sofa::component::forcefield::TriangularFEMForceField<sofa::defaulttype::Vec3Types>::TriangleInformation
16>        ]
```

Reply:
And what about the RAII the longer version with 
WriteOnlyAccessor, does it work ?

Reply:
helper::WriteOnlyAccessor< topology::TriangleData<sofa::type::vector<TriangleInformation> > > triangleInf = triangleInfo;
gives:
```
'sofa::component::topology::TopologyData<sofa::topology::Triangle,sofa::type::vector<sofa::component::forcefield::TriangularFEMForceField<sofa::defaulttype::Vec3Types>::TriangleInformation,sofa::type::CPUMemoryManager<T>>>' to 'sofa::helper::WriteOnlyAccessor<sofa::component::topology::TopologyData<sofa::topology::Triangle,sofa::type::vector<sofa::component::forcefield::TriangularFEMForceField<sofa::defaulttype::Vec3Types>::TriangleInformation,sofa::type::CPUMemoryManager<T>>>,void>'	SofaMiscFem	C:\projects\sofa-src\modules\SofaMiscFem\src\SofaMiscFem\TriangularFEMForceField.inl	166	
Error	C2676	binary '[': 'sofa::helper::WriteOnlyAccessor<sofa::component::topology::TopologyData<sofa::topology::Triangle,sofa::type::vector<sofa::component::forcefield::TriangularFEMForceField<sofa::defaulttype::Vec3Types>::TriangleInformation,sofa::type::CPUMemoryManager<T>>>,void>' does not define this operator or a conversion to a type acceptable to the predefined operator	SofaMiscFem	C:\projects\sofa-src\modules\SofaMiscFem\src\SofaMiscFem\TriangularFEMForceField.inl	168	
```